pkg/logql/engine.go

```diff
@@ -402,6 +411,23 @@ func (q *query) evalSample(ctx context.Context, expr syntax.SampleExpr) (promql_
 	return result, stepEvaluator.Error()
 }
 
+func (q *query) checkLimits(expr syntax.SampleExpr, limit time.Duration) error {
+	var err error
+	expr.Walk(func(e interface{}) {
+		switch e.(type) {
+		case *syntax.RangeAggregationExpr:
+			l := e.(*syntax.RangeAggregationExpr).Left
+			if l != nil {
+				if l.Interval > limit {
+					err = fmt.Errorf("%w: [%s] > [%s]", logqlmodel.ErrRangeLimit, l.Interval, limit)
+				}
+			}
+			return
+		}
```

Comment:
could be simplified with :
```suggestion
		switch e := e.(type) {
		case *syntax.RangeAggregationExpr:
			if e.Left == nil || e.Left.Interval <= limit {
				return
			}
			err = fmt.Errorf("%w: [%s] > [%s]", logqlmodel.ErrRangeLimit, e.Left.Interval, limit)
		}
```